browser/ui/webui/brave_education/education_page_handler.cc

```diff
@@ -0,0 +1,121 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/webui/brave_education/education_page_handler.h"
+
+#include "base/containers/fixed_flat_set.h"
+#include "brave/browser/ai_chat/ai_chat_utils.h"
+#include "brave/browser/brave_rewards/rewards_service_factory.h"
+#include "brave/browser/brave_wallet/brave_wallet_service_factory.h"
+#include "brave/components/brave_education/common/education_urls.h"
+#include "brave/components/brave_vpn/common/buildflags/buildflags.h"
+#include "brave/components/constants/webui_url_constants.h"
+#include "chrome/browser/profiles/profile.h"
+
+#if BUILDFLAG(ENABLE_BRAVE_VPN)
+#include "brave/browser/brave_vpn/vpn_utils.h"
+#endif
+
+namespace brave_education {
+
+namespace {
+
+bool IsCommandSupportedForPageType(mojom::Command command,
+                                   EducationPageType page_type) {
+  switch (page_type) {
+    case EducationPageType::kGettingStarted: {
+      static constexpr auto kCommands = base::MakeFixedFlatSet<mojom::Command>(
+          {mojom::Command::kOpenRewardsOnboarding,
+           mojom::Command::kOpenWalletOnboarding,
+           mojom::Command::kOpenVPNOnboarding, mojom::Command::kOpenAIChat});
+      return kCommands.contains(command);
+    }
+  }
+  NOTREACHED();
+}
+
+bool CanShowWalletOnboarding(Profile* profile) {
+  return brave_wallet::BraveWalletServiceFactory::GetServiceForContext(
+             profile) != nullptr;
+}
+
+bool CanShowRewardsOnboarding(Profile* profile) {
+  return brave_rewards::RewardsServiceFactory::GetForProfile(profile) !=
+         nullptr;
+}
+
+bool CanShowVPNBubble(Profile* profile) {
+#if BUILDFLAG(ENABLE_BRAVE_VPN)
+  return brave_vpn::IsAllowedForContext(profile);
+#else
+  return false;
+#endif
+}
+
+bool CanShowAIChat(Profile* profile) {
+  return ai_chat::IsAllowedForContext(profile);
```

Comment:
@petemill can we get to get rid of `IsAllowedForContext`? It isn't needed now that AIChat is a service. @zenparsing here and above in vpn we should check the service for null

Reply:
I've change it so that we check for the AI chat server. For the VPN check, I'll have to look some more, as unit tests will not work unless a `BraveVPNConnectionManager` is set up for the process, as [here](https://github.com/brave/brave-core/blob/ab9b1b06c4a25c8981421b3aca24f87f19acb00d/browser/ui/views/toolbar/brave_vpn_button_unittest.cc#L75), and this requires too much knowledge about the VPN feature. Perhaps there should be a testing helper in the VPN code.

Reply:
where does that normally get set up?

Reply:
I don't know.